src/components/pages/AppDetail/components/AppDetailHeader/index.tsx

```diff
@@ -92,11 +86,12 @@ export default function AppDetailHeader({ item }: AppDetailHeaderProps) {
         break
       default:
         btnColor = {
-          color:
-            user.roles.indexOf(Roles.SUBSCRIBE_APPS) !== -1 &&
-            user.roles.indexOf(Roles.SUBSCRIBE_SERVICE) !== -1
-              ? 'primary'
-              : 'secondary',
+          color: userHasPortalRole([
+            ROLES.SUBSCRIBE_APPS,
+            ROLES.SUBSCRIBE_SERVICE,
+          ])
```

Comment:
@jjeroch Not sure what this is good for but I think it was implemented incorrectly setting the button color only if the user has both roles. Now the check is from AND to OR setting the color if the user has at least one of the two roles.

Reply:
@jjeroch yes, there is the same check further below if the button is selectable or not